CMakeLists.txt

```diff
@@ -401,33 +401,42 @@ get_directory_property( DirDefs COMPILE_DEFINITIONS )
 # Configuration summary
 ################################################################
 
-message("== Configuration Summary ==")
-message("  PROJECT_SOURCE_DIR:   ${PROJECT_SOURCE_DIR}")
-message("  PROJECT_BINARY_DIR:   ${PROJECT_BINARY_DIR}")
-message("  CMAKE_INSTALL_PREFIX: ${CMAKE_INSTALL_PREFIX}")
-message("  CMAKE_BUILD_TYPE:     ${CMAKE_BUILD_TYPE}")
+set(_str "== Configuration Summary ==\n")
```

Comment:
Please put in a note in the CMake file so that people know why you are using strings, not the message function.  Also, could we use the messsage(STATUS instead of the string to keep it closer to a standard approach?